js/adapt-assessmentArticleModel.js

```diff
@@ -703,18 +701,10 @@ define([
       var attemptInProgress = restoreState[4] == 1 ? true : false;
       var scoreAsPercent;
 
-      var indexByIdQuestions = restoreState[5];
+      let blocks = blockData[0].map(trackingId => Adapt.data.findWhere({ _trackingId: trackingId }));
+      let restoredChildrenModels = blocks;
 
-      var blockIds = {};
-      for (id in indexByIdQuestions) {
-        if (indexByIdQuestions.hasOwnProperty(id)) {
-          var blockId = Adapt.findById(id).get('_parentId');
-          blockIds[blockId] = Adapt.findById(blockId);
-        }
-      }
-      var restoredChildrenModels = _.values(blockIds);
-
-      if (indexByIdQuestions) this.getChildren().models = restoredChildrenModels;
+      if (restoredChildrenModels && restoredChildrenModels.length) this.getChildren().models = restoredChildrenModels;
```

Comment:
Minor: Probably nicer to have this across multiple lines

Reply:
Does `restoredChildrenModels` even have to be checked as a truthy? `.map()` should always return an array.

Reply:
Notwithstanding the fact that this variable isn't even needed in the new code 😉.